src/dialog-controller.js

```diff
@@ -58,7 +58,9 @@ export class DialogController {
    * @returns Promise An empty promise object.
    */
   close(ok: boolean, output?: any): Promise<DialogResult> {
-    if (this._closePromise) return this._closePromise;
+    if (this._closePromise) { 
+      return this._closePromise;
+    }
 
     this._closePromise = invokeLifecycle(this.viewModel, 'canDeactivate').then(canDeactivate => {
```

Comment:
I'm not sure what should be done if the promise returned from `invokeLifecycle` is rejected. Should `_closePromise` still be cleared. Or call `.error()`?